src/test/scala/org/globalforestwatch/summarystats/annualupdate_minimal/AnnualUpdateMinimalSpec.scala

```diff
@@ -0,0 +1,147 @@
+      .options(AnnualUpdateMinimalExport.csvOptions)
+      .csv(path = outputPath)
+  }
+  
+  def readAumResult(path: String) = {
+    spark.read
+      .options(Map(
+        "header" -> "true",
+        "delimiter" -> "\t",
+        "escape" -> "\"",
+        "quoteMode" -> "MINIMAL",
+        "nullValue" -> "",
+        "emptyValue" -> "",
+        "inferSchema" -> "true"
+      ))
+      .csv(path)
+  }
+
+  it("matches recorded output for GADM", DefaultTag) {
+    val gadmDF: DataFrame = AnnualUpdateMinimal(idn1_5GadmInputTsvPath, "gadm")
+
+    // Transform results to match expected output
+    import spark.implicits._
+    val unpackedDF = gadmDF      
+      .transform(
+        AnnualUpdateMinimalDF.unpackValues(
+          List($"id.iso" as "iso", $"id.adm1" as "adm1", $"id.adm2" as "adm2")
+        )
+      )
+    val exportDF: DataFrame = unpackedDF.transform(AnnualUpdateMinimalDF.aggSummary(List("iso", "adm1", "adm2"))) 
+    val top20Rows = exportDF.limit(20)
+    
+    // Export and save results
+    top20Rows
+      .write
+      .options(csvOptions)
+      .csv("output/gadm-aum-output")
+
+    // Uncomment to save new expected results
+    //saveExpectedAumResult(top20Rows, idn1_5GadmExpectedOutputPath)
+
+    // Read expected results and compare
+    val expectedDF = readAumResult(idn1_5GadmExpectedOutputPath)
+    val top20RowsDF = readAumResult("output/gadm-aum-output")
+
+    assertApproximateDataFrameEquality(top20RowsDF, expectedDF, 0.00001, ignoreNullable = true)
+
+  }
+
+  it("matches recorded output for WDPA", DefaultTag) {
+    val wdpaDF: DataFrame = AnnualUpdateMinimal(wdpaInputTsvPath, "wdpa")
+
+    // Transform results to match expected output
+    import spark.implicits._
+    val idCols: List[String] = List(
+      "wdpa_protected_area__id",
+      "wdpa_protected_area__name",
+      "wdpa_protected_area__iucn_cat",
+      "wdpa_protected_area__iso",
+      "wdpa_protected_area__status"
+    )
+    val unpackedDF = wdpaDF      
+      .transform(
+        AnnualUpdateMinimalDF.unpackValues(
+          List(
+            $"id.wdpaId" as "wdpa_protected_area__id",
+            $"id.name" as "wdpa_protected_area__name",
+            $"id.iucnCat" as "wdpa_protected_area__iucn_cat", 
+            $"id.iso" as "wdpa_protected_area__iso",
+            $"id.status" as "wdpa_protected_area__status"
+          ),
+          wdpa = true
+        )
+      )
+    val exportDF = unpackedDF.transform(AnnualUpdateMinimalDF.aggSummary(idCols, wdpa=true))
+    val top20Rows = exportDF.limit(20)
+
+    // Export and save results
+    top20Rows
+      .write
+      .options(csvOptions)
+      .csv("output/wdpa-aum-output")
+
+    // Uncomment to save new expected results
+    //saveExpectedAumResult(top20Rows, wdpaExpectedOutputPath)
+
+    // Read expected results and compare
+    val expectedDF = readAumResult(wdpaExpectedOutputPath)
+    val top20RowsDF = readAumResult("output/wdpa-aum-output")
```

Comment:
Do you still actually need to write top20Rows out and then read it back in to make the test work?  If so, can you add a comment as to why you are doing it (something about changing the format of the null/empty fields, I think?)

Reply:
My question was not about why you are only using the top 20 rows, but why you are writing out top20Rows and reading it back in before you compare.  That is, why can't you use top20Rows in your comparison, rather than writing it out, reading it back into top20RowsDF and then doing the comparison?

Would be good to put a comment here about why you're doing that, but you can leave that for some other time if you want.